src/schemas/json/appsettings.json

```
@@ -90,16 +91,241 @@
           "pattern": "^[0-9a-f]{32}$"
         }
       },
-      "required": ["ApiKey", "LogId"]
+      "required": [
+        "ApiKey",
+        "LogId"
+      ]
+    },
+    "protocols": {
+      "description": "The protocols enabled on the endpoint.",
+      "type": "string",
+      "enum": [
+        "None",
+        "Http1",
+        "Http2",
+        "Http1AndHttp2",
+        "Http3",
+        "Http1AndHttp2AndHttp3"
+      ]
+    },
+    "certificate": {
+      "description": "Certificate configuration.",
+      "type": "object",
+      "properties": {
+        "Path": {
+          "description": "The certificate file path. If a file path is specified then the certificate will be loaded from the file system.",
+          "type": "string"
+        },
+        "KeyPath": {
+          "description": "The certificate key file path.",
+          "type": "string"
+        },
+        "Password": {
+          "description": "The certificate password used to access the private key.",
+          "type": "string"
+        },
+        "Subject": {
+          "description": "The certificate subject. If a subject is specified then the certificate will be loaded from the certificate store.",
+          "type": "string"
+        },
+        "Store": {
+          "description": "The certificate store name. Defaults to 'My'.",
+          "type": "string",
+          "default": "My"
+        },
+        "Location": {
+          "description": "The certificate store location. Defaults to 'CurrentUser'.",
+          "type": "string",
+          "enum": [
+            "LocalMachine",
+            "CurrentUser"
+          ],
+          "default": "CurrentUser"
+        },
+        "AllowInvalid": {
+          "description": "A value indicating whether or not to load certificates that are considered invalid. Defaults to false.",
+          "type": "boolean",
+          "default": false
+        }
+      }
+    },
+    "sslProtocols": {
+      "description": "Specifies allowable SSL protocols. Defaults to 'None' which allows the operating system to choose the best protocol to use, and to block protocols that are not secure. Unless your app has a specific reason not to, you should use this default.",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": [
+          "None",
+          "Ssl2",
+          "Ssl3",
+          "Tls",
+          "Default",
+          "Tls11",
+          "Tls12",
+          "Tls13"
```

Can we de-list values that are strongly not recommended/obsolete?
```suggestion
          "None",
          "Ssl3",
          "Tls11",
          "Tls12",
          "Tls13"
```

---

The answer is to just leave them out. Then they won't show up in intelli-sense. So should SslProtocols be removed?

---

@halter73 may have some suggestions on the exact values, but the above is my short list.

---

We could go with all the non-obsoleted SslProtocols which is everything @Tratcher listed plus "Tls" minus "Ssl3".

```C#
    [System.FlagsAttribute]
    public enum SslProtocols
    {
        None = 0,
        [System.ObsoleteAttribute("This value has been deprecated.  It is no longer supported. https://go.microsoft.com/fwlink/?linkid=14202")]
        Ssl2 = 12,
        [System.ObsoleteAttribute("This value has been deprecated.  It is no longer supported. https://go.microsoft.com/fwlink/?linkid=14202")]
        Ssl3 = 48,
        Tls = 192,
        [System.ObsoleteAttribute("This value has been deprecated.  It is no longer supported. https://go.microsoft.com/fwlink/?linkid=14202")]
        Default = 240,
        Tls11 = 768,
        Tls12 = 3072,
        Tls13 = 12288,
    } 
```

https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Net.Primitives/ref/System.Net.Primitives.cs#L509-L523

---

Nice, I'd forgotten they'd actually obsoleted some of them.

---

Done. Is the list correct now?

```json
"enum": [
  "None",
  "Ssl3",
  "Tls11",
  "Tls12",
  "Tls13"
],
```

---

```
"enum": [
  "None",
  "Tls",
  "Tls11",
  "Tls12",
  "Tls13"
],
```